public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -293,3 +297,34 @@ export const getLokiQueryFromDataQuery = (query?: DataQuery): LokiQuery | undefi
 
   return query;
 };
+
+export function formatLogqlQuery(query: string, datasource: LokiDatasource) {
+  reportInteraction('grafana_loki_format_query_clicked', {});
```

Comment:
I think we could also add some extra formation here like if it's a logs query, metric query, query with errors.

Reply:
For example, if they press it when they have errors often, which we don't support, we might do something about it, e.g. messages, etc.